### DISCUSSION TITLE
what is the meaning about  computeFrameJacobian?

### DISCUSSION THREAD

**Discussion by laniakeawhite** [stack-of-tasks/pinocchio#2113] [Q&A] [answered]
According to the book <MODERN ROBOTICS,MECHANICS, PLANNING, AND CONTROL>,there are two types jacobian:  space jacobian and body jacobian.

If i  pass the end effector frame id  of robot arm to  function **computeFrameJacobian**,does it  calculate the body jacobian about the end effector?

**Comment by stephane-caron** [comment, marked as answer]
Looking at the [documentation of `computeFrameJacobian`](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/namespacepinocchio.html#a7b1189aea93a252f4c06d9a7e3c9f5e7), the fifth argument to the function is a [`ReferenceFrame`](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/group__pinocchio__multibody.html#gaa56acd11081bc4603c21de42b9c46c81) enumeration that can take the following values:

- `WORLD`: The WORLD frame convention corresponds to the frame concident with the Universe/Inertial frame but moving with the moving part (Joint, Frame, etc.).
- `LOCAL`: The LOCAL frame convention corresponds to the frame directly attached to the moving part (Joint, Frame, etc.) where the coordinates basis matches the local coordinates system associated with the moving part. It also called the BODY representation in the litterature.

The `WORLD` frame convention yields the space Jacobian, while the `LOCAL` frame yields the body Jacobian.

**Reply by stephane-caron** [reply]
You can also read more about these conventions in the section **Frame options in Pinocchio** of [this lab notebook](https://github.com/ymontmarin/_tps_robotique/blob/31e9661d8352c7436ede5de1149dbf966cc86fcc/3_invkine.ipynb).

**Reply by laniakeawhite** [reply]
thx!
